library/std/src/sys/zkvm/thread_local_key.rs

```diff
@@ -0,0 +1,23 @@
+use crate::alloc::{alloc, Layout};
+
+pub type Key = usize;
+
+#[inline]
+pub unsafe fn create(_dtor: Option<unsafe extern "C" fn(*mut u8)>) -> Key {
+    alloc(Layout::new::<*mut u8>()) as _
+}
+
+#[inline]
+pub unsafe fn set(key: Key, value: *mut u8) {
+    let key: *mut *mut u8 = core::ptr::from_exposed_addr_mut(key);
```

Comment:
This is changed from the previous `as *mut *mut` which was removed due to: https://github.com/rust-lang/rust/issues/95228 I think using `from_exposed_addr_mut` gets us the same feature but im not 100% on it.